src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,143 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+    private static final int ZERO_HASH = 0;
+
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putNullKey(value);
+            return;
+        }
+
+        int hash = key.hashCode();
+        int index = getIndex(hash);
 
+        Entry<K, V> current = table[index];
+        while (current != null) {
+            if (current.key != null && current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            current = current.next;
+        }
+
+        Entry<K, V> newEntry = new Entry<>(key, value);
+        newEntry.next = table[index];
+        table[index] = newEntry;
+        size++;
+
+        if ((double) size / table.length > LOAD_FACTOR) {
+            resizeTable();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return getNullVal();
+        }
+
+        int hash = key.hashCode();
+        int index = getIndex(hash);
+
+        Entry<K, V> current = table[index];
+        while (current != null) {
+            if (current.key != null && current.key.equals(key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+        private Entry<K, V> next;
+
+        public Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+    }
+
+    private int getIndex(int hash) {
+        return (hash == ZERO_HASH) ? ZERO_HASH : Math.abs(hash % table.length);
+    }
+
+    private int getIndex(int hash, int tableSize) {
+        return Math.abs(hash % tableSize);
+    }
```

Comment:
i think you can refactor your code so you can use only one getIndex, try it :)
Maybe the key to that would be reusing put method in resize method